src/Generator/Views/BaseJsonFormatter.Generated.cshtml

```diff
@@ -0,0 +1,171 @@
+            bool ReadRef@(Raw("<T>")) (ref JsonReader reader, ref T set)
+            {
+                var formatter = formatterResolver.GetFormatter@(Raw("<T>"))();
+                set = formatter.Deserialize(ref reader, formatterResolver);
+                return true;
+            }
+            bool Read@(Raw("<T>")) (ref JsonReader reader, Base b, Action@(Raw("<Base, T>")) set)
+            {
+                var formatter = formatterResolver.GetFormatter@(Raw("<T>"))();
+                set(b, formatter.Deserialize(ref reader, formatterResolver));
+                return true;
+            }
+            static bool ReadString(ref JsonReader reader, ref string stringProp)
+            {
+                stringProp = reader.ReadString();
+                return true;
+            }
+            
+            while (reader.ReadIsInObject(ref count))
+            {
+                var propertyName = reader.ReadPropertyNameSegmentRaw();
+                if (AutomataDictionary.TryGetValue(propertyName, out var value))
+                {
+                    @{ index = 0; }
+                    _ = value switch
+                    {
+@foreach (var field in Model.BaseFieldsOrdered)
+{
+    if (field.Item1 != "WriteProp" && field.Item2.ClrType == "string")
+{<text>                        @(index++) => ReadString(ref reader, ref @(field.Item2.Name.Replace(".",""))),
+</text>}
+    else if (field.Item1 != "WriteProp")
+{<text>                        @(index++) => ReadRef<@(Raw(field.Item2.ClrType))>(ref reader, ref @(field.Item2.Name.Replace(".",""))),
+</text>}
+    else
+{<text>                        @(index++) => Read<@(Raw(field.Item2.ClrType))>(ref reader, ecsEvent, (b, v) => b.@(FileGenerator.PascalCase(field.Item2.Name)) = v),
+</text>}
+}
+@foreach (var schema in Model.YamlSchemasOrdered.Skip(1))
+{<text>                        @(index++) => Read<@(FileGenerator.PascalCase(schema.Name))>(ref reader, ecsEvent, (b, v) => b.@(FileGenerator.PascalCase(schema.Name)) = v),
+</text>}
+                        _ => false
+                    };
+                }
+            }
+            ecsEvent.Log ??= new Log();
+            ecsEvent.Log.Level = loglevel;
+            ecsEvent.Message = message;
+            ecsEvent.Timestamp = timestamp;
+            return ecsEvent;
+        }
+
+        public void Serialize(ref JsonWriter writer, Base value, IJsonFormatterResolver formatterResolver)
+        {
+            if (value == null) return;
+            writer.WriteBeginObject();
+@{ index = 0; }
+            // Base fields
+@foreach (var field in Model.BaseFieldsOrdered)
+{
+    if (field.Item1 != "WriteProp")
+    {<text>            @(field.Item1)(ref writer, value, formatterResolver); // @(index++)
+</text>}
+    else
+    {<text>            @(field.Item1)(ref writer, "@field.Item2.Name", value.@FileGenerator.PascalCase(field.Item2.Name), formatterResolver); // @(index++)
+</text>}
+}
+            // Complex types
```

Comment:
Love this but since this is the only bit that needs generation perhaps we can rely on this class being partial and we generate a method stub we can call from the manually implemented side of this formatter?